src/NHibernate/Persister/Entity/AbstractEntityPersister.cs

```diff
@@ -2350,21 +2391,19 @@ private void InternalInitSubclassPropertyAliasesMap(string path, IEnumerable<Pro
 					Component component = (Component)prop.Value;
 					InternalInitSubclassPropertyAliasesMap(propName, component.PropertyIterator);
 				}
-				else
-				{
-					string[] aliases = new string[prop.ColumnSpan];
-					string[] cols = new string[prop.ColumnSpan];
-					int l = 0;
-					foreach (ISelectable thing in prop.ColumnIterator)
-					{
-						aliases[l] = thing.GetAlias(Factory.Dialect, prop.Value.Table);
-						cols[l] = thing.GetText(Factory.Dialect);
-						l++;
-					}
 
-					subclassPropertyAliases[propName] = aliases;
-					subclassPropertyColumnNames[propName] = cols;
+				var aliases = new string[prop.ColumnSpan];
+				var cols = new string[prop.ColumnSpan];
+				var l = 0;
+				foreach (var thing in prop.ColumnIterator)
+				{
+					aliases[l] = thing.GetAlias(Factory.Dialect, prop.Value.Table);
+					cols[l] = thing.GetText(Factory.Dialect);
+					l++;
 				}
+
+				subclassPropertyAliases[propName] = aliases;
```

Comment:
Now `subclassPropertyAliases` contains also component properties which is needed in order to select component properties from a subquery.